src/main/java/hudson/plugins/active_directory/ActiveDirectorySecurityRealm.java

```diff
@@ -600,7 +609,25 @@ public DirContext createDNSLookupContext() throws NamingException {
                     }
 
                     public int compareTo(PrioritizedSocketInfo that) {
-                        return that.priority-this.priority; // sort them so that bigger priority comes first
+                        int prio = that.priority - this.priority; // sort them so that bigger priority comes first
+                        if (prio != 0) {
+                            return prio;
+                        }
+                        String s1 = socket != null ? socket.toString() : "";
+                        String s2 = that.socket != null ? that.socket.toString() : "";
+                        return s1.compareTo(s2);
```

Comment:
:bug:  Sorting entries of the same priority by host name is unwise, as it forces all the clients to go through the same order and fails to balance the load. We should just stick to the priority sorting and the `socket` field shouldn't enter into the comparison.

hashCode and equal functions are dead code and not needed here, so I'd rather not have them for brevity, but maybe that's more of a matter of taste.